clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1683,10 +1683,10 @@ private Fetch<K, V> collectFetch() {
      * @throws NoOffsetForPartitionException If no offset is stored for a given partition and no offset reset policy is
      *                                       defined
      */
-    private boolean updateFetchPositions(final Timer timer) {
+    private boolean updateFetchPositions(final Timer timer, final boolean isPassedByErrorEvent) {
```

Comment:
if the suggestion below works and we can get rid of the ErrorEvent for metadata errors, I hope we could end up removing this param, and just having a kind of requiresSubscriptionMetadata defined in events, and the CheckAndUpdatePositions event would have that true. So the network client would know that it needs to completeExceptionally the future for that event (and not for others that are not related to subscription metadata)....just an idea that depends on whether we can simplify the metadata error event